source/beamline/SXRMB/SXRMBFourElementVortexDetector.cpp

```diff
@@ -55,6 +76,75 @@ QString SXRMBFourElementVortexDetector::synchronizedDwellKey() const
+	detectorInitializationActions->addSubAction(AMXRFDetector::createInitializationActions());
+
+	AMListAction3 *scanPVInitializationListActions = new AMListAction3(new AMListActionInfo3("Votex Detector Scan PV Initialization", "Votex Detector Scan PV Initialization Actions"), AMListAction3::Parallel);
+	detectorInitializationActions->addSubAction(scanPVInitializationListActions);
+
+	AMListAction3 *scanPVInitializationActions = 0;
+
+	// add the MCA initialization actions
+	scanPVInitializationActions = new AMListAction3(new AMListActionInfo3("Votex MCA Rate initialization", "Votex MCA Rate initialization"), AMListAction3::Sequential);
+	scanPVInitializationActions->addSubAction(AMActionSupport::buildControlMoveAction(mcaRatePVControl_, 0));  // passive
+	scanPVInitializationActions->addSubAction(AMActionSupport::buildControlWaitAction(mcaRatePVControl_, 0));
+
+	scanPVInitializationListActions->addSubAction(scanPVInitializationActions);
+
+	// add the DXPs initialization actions
+	scanPVInitializationActions = new AMListAction3(new AMListActionInfo3("Votex DXPs Rate initialization", "Votex DXPs Rate initialization"), AMListAction3::Sequential);
+	scanPVInitializationActions->addSubAction(AMActionSupport::buildControlMoveAction(dxpRatePVControl_, 0));  // passive
+	scanPVInitializationActions->addSubAction(AMActionSupport::buildControlWaitAction(dxpRatePVControl_, 0));
+
+	scanPVInitializationListActions->addSubAction(scanPVInitializationActions);
+
+	// add the status initialization actions
+	scanPVInitializationActions = new AMListAction3(new AMListActionInfo3("Votex Staus Rate initialization", "Votex Status Rate initialization"), AMListAction3::Sequential);
+	scanPVInitializationActions->addSubAction(AMActionSupport::buildControlMoveAction(statusRatePVControl_, 9));  // 0.1 second
+	scanPVInitializationActions->addSubAction(AMActionSupport::buildControlWaitAction(statusRatePVControl_, 9));
+
+	scanPVInitializationListActions->addSubAction(scanPVInitializationActions);
+
+
+	return detectorInitializationActions;
+}
+
+AMAction3* SXRMBFourElementVortexDetector::createCleanupActions()
+{
+	AMListAction3 *detectorCleanupListActions = new AMListAction3(new AMListActionInfo3("4E Votex Detector Cleanup", "4E Votex Detector Cleanup Actions"), AMListAction3::Sequential);
+	detectorCleanupListActions->addSubAction(AMXRFDetector::createCleanupActions());
+
+	AMListAction3 *scanPVCleanupListActions = new AMListAction3(new AMListActionInfo3("Votex Detector Scan PV Cleanup", "Votex Detector Scan PV Cleanup Actions"), AMListAction3::Parallel);
+	detectorCleanupListActions->addSubAction(scanPVCleanupListActions);
+
+	AMListAction3 *scanPVCleanupActions = 0;
+
+	// add the MCA cleanup actions
+	scanPVCleanupActions = new AMListAction3(new AMListActionInfo3("Votex MCA cleanup", "Votex MCA cleanup"), AMListAction3::Sequential);
+	scanPVCleanupActions->addSubAction(AMActionSupport::buildControlMoveAction(mcaRatePVControl_, mcaRatePVControl_->value()));
```

Comment:
For scan cleanup, we suppose to set the setpoint to whatever it was before the scan. That's the reason we use the pv->value() here instead of a variable.